src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorSyntaxNodeExtensions.cs

```diff
@@ -270,8 +270,17 @@ static LinePosition GetLinePosition(SourceLocation location)
         }
     }
 
-    public static SyntaxNode? FindInnermostNode(this SyntaxNode node, int index, bool includeWhitespace = false)
+    /// <summary>
+    /// Finds the innermost SyntaxNode for a given location in source, within a given node.
+    /// </summary>
+    /// <param name="node">The parent node to search inside.</param>
+    /// <param name="index">The location to find the innermost node at.</param>
+    /// <param name="includeWhitespace">Whether to include whitespace in the search.</param>
+    /// <param name="walkMarkersBack">When true, if there are multiple <see cref="SyntaxKind.Marker"/> tokens in a single location, return the parent node of the
```

Comment:
I'm not sure I know what a Marker is, or when this would be false? Do lots of tests fail if the default is changed to true (or we just remove the parameter?)

Reply:
Markers are basically "there's a construct here, but no tokens, editors may want to know about this fact." From my testing, you'd only ever see them when you ask for whitespace, though it's possible I'm missing something there.

Reply:
That being said, I think we can probably loosen the restrictions and turn it on by default. It's almost certainly the right behavior for most code paths in the IDE: if there are multiple possible constructs at a single location, it's very likely that the first construct is the one you want to know about.